src/text-input/src/TextInput.js

```diff
@@ -89,8 +106,11 @@ class TextInput extends PureComponent {
     const borderRadius = theme.getBorderRadiusForControlHeight(height)
 
     return (
-      <Text
-        is="input"
+      <StatelessTextInput
+        value={this.state.value}
+        isClearable={isClearable}
+        onClear={() => this.setState({ value: '' })}
+        onChange={e => this.setState({ value: e.target.value })}
```

Comment:
Rebinding new functions on every render will defeat the purpose of using PureComponent. I would extract these to bound functions:

```js
class Foo {
  myFunction = (event) => {}
}
```

Reply:
My bad! -- fixed this bit (I think)